eslint.config.js

```diff
@@ -34,6 +34,32 @@ export default tsEslint.config(
       "@typescript-eslint/consistent-type-exports": "error",
       "@typescript-eslint/no-import-type-side-effects": "error",
       "@typescript-eslint/consistent-type-imports": "error",
+      "@typescript-eslint/use-unknown-in-catch-callback-variable": "off",
+      "@typescript-eslint/restrict-template-expressions": [
+        "error",
+        {
+          allowAny: true,
+          allowBoolean: true,
+          allowNullish: false,
+          allowNumber: true,
+          allowRegExp: true,
+        },
+      ],
```

Comment:
This rule I catches some nice little bugz that happen inside template literals, like `[object] [object]` and encourages you to provide fallbacks for nullish values.

Template literals will call the `.toString()` method on everything you put there, so with that understanding, I think it is fine to allow the above values, as `toString()` was not adding any value having that enforced.
However, it did make me consider numbers, as most numbers can be handled fine by `toString()`, big numbers in UI should be displayed with `.toLocaleString()`,  and any floating point math should be dealt with before you end up with ugly `3.0000000000000001`.

I kept the `allowNullish: false` however. The more time that goes by, the more I'm bothered by my logs that end up with: `"error with item id:"` where the item didn't exist or something. This rule would ask the developer to provide some feedback, like 
```ts
`error with item id: ${item?.id ?? "<item not found>"}`
```

https://typescript-eslint.io/rules/restrict-template-expressions/